numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -621,6 +594,31 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_str(PyObject *self){
+    PyObject *arr, *ret = NULL;
+
+    if (!PyDataType_HASFIELDS(((PyVoidScalarObject*)self)->descr)) {
+        PyObject  *item, *item_str;
+
+        item = gentype_generic_method(self, NULL, NULL, "item");
+        if (item == NULL) {
+            return NULL;
+        }
+
+        item_str = PyObject_Str(item);
+        Py_DECREF(item);
+        return item_str;
+    }
```

Comment:
I think we're stuck with `item` -- the whole point of it is to just return python things.